### ISSUE TITLE
Some Tx payload's failed to deserialise

### ISSUE THREAD

**Issue by stake1edge** [bloxbean/cardano-client-lib#153]
Hi, In both 0.2.0 & 0.3.0-beta2, for certain transaction payloads seeing this:

> Caused by: com.bloxbean.cardano.client.exception.CborDeserializationException: Cbor deserialization failed. Invalid type. BREAK
OR
> Caused by: com.bloxbean.cardano.client.exception.CborDeserializationException: Invalid type for TransactionOutput : SPECIAL

Details:
- Error occurs when doing this: `Transaction tx = Transaction.deserialize(HexUtil.decodeHexString( <hex encoded bytes[] payload> ));`
- The error "invalid type BREAK" as far as I can tell are all MINSWAP (v2) contract related
- encoded payloads in examples below were retrieved using Carp

Two examples:
**- TX Hash: 86bc85d7d492e1ae91a7a84ef8d17e84d5c848276b7b0c825476a3627a74df4f**
   Payload: 84a90082825820a4c0024af0af176f36322c151bdcb9e8ac482cb815cf0b238a464604dfa3222301825820942d3c5e5e1bf6a62fdcd460b0637e11fd8b16c8ab506dc1d46d95e8ab9f2aff02018382583901adb1bf6a51b20ff1b8450726ef3891bb0e153d5bf47783375e2134afbd6a096cbba5e259946798e948403e2d2b3d9ea88a12ee8e7ae94497821a002583d6a1581cd195ca7db29f0f13a00cac7fca70426ff60bad4e1e87d3757fae8484a24568764144411a00136fbe4568764d494e1a10adbdd283581d719b85d5e8611945505f078aeededcbed1d6ca11053f61e3f9d999fe44821a001f0b06a2581cd195ca7db29f0f13a00cac7fca70426ff60bad4e1e87d3757fae8484a14a3138313330303239343001581ce4214b7cce62ac6fbba385d164df48e157eae5863521b4b67ca71d86a158201b7f4abbf3eb04f8a7e5fbbc2042c524210dd960b6703a02fe52f70a7701e2841a02f3e658582016374a6aa3b689334db7e20e58d3aa4a28fe345580922ae876eb162d8b6b0ab882583901030c012b604482130d1f821d2ec39f76218cd89c00a3fa7662fbe7b5ef7879c68377187aad9335d004f3abb73dd109f6f674877b367c246a1a81fe000b021a00093565031a0440797b0758203dc4560b00c5a65ff6836aba7bde54caf9f0dda2471e9305983f5da114007a7609a1581cd195ca7db29f0f13a00cac7fca70426ff60bad4e1e87d3757fae8484a44568764144411a00136fbe4568764d494e1a10adbdd24a31353333313831313632204a31383133303032393430010b5820aa861d32a3d18c9159fa25b9b0a18a6bd0e9c7b63915b9b14930ef132482495e0d81825820d3d8280f39c3152c91d86e741631075d50043a60d90988c42a7b42a7f182c459000e81581c030c012b604482130d1f821d2ec39f76218cd89c00a3fa7662fbe7b5a500838258205424fa10ba83c95c33714c420479c19183a7274e7c1d4161d173842c245b340c584037e731bb8d9c8f684a561c873bef660c2e1b1205d0b8f2d52573fb5b56ff8b12a08a6e5d55bef01cb669fe41d8e0ead9511274f19d99121624599dbced3c520e825820671f554b5fb1caa16485732ef7c28a3b046461e4da0a30bcbeedf73fcdc32c96584092594a6d8c52cbc0788d4efc932ae83a763cfbdd37a17efd933130c73ab38abbe55f2cd630d8955f842581aa4c1d683b56091e77ddceffdfa639f3557bd1d308825820c3ef85f3614983e752677203b942a63ba76b52e8fbca4eb37d2b1ac4106237d658409399f564afc4ee4a663858559bc7137d427ebd5d75213ebc109b0fc1f066580b1095aa55b9b2558de1801ffc09fb3e6227aa19059784dc2fccb374380aee640d01818200581c4f641455f17911fe2f55ad3ad67fc2e0b2946b59af3352574322e67e03815906f45906f10100003232323232323232323232323232323232323232323232323232323232323232323232232223232323232323553353019003210011622323235301b35301a3500202c01922325335533553335734606800204c2a666ae68c0cc0040540c454cd4ccd54c0b00b4c05804cc0a14040cc08dd71aa80491111110118a99a980419980c3ae5017355335301912233301c22533533018301f002500813300400200110010013302501f35500b034162215335001100222160121300833301875ca02e6a603a6a00a05e02404c04c2a66a666aa605805a602c0266050a02066046eb8d54024888888808c54cd54cd4c064488ccc070894cd4cc060c07c00940204cc0100080044004004cc09407cd5402c0d00548854cd40040a0880644ccd5cd19b87355335300d33301875ca02e6a603a6a00a05e02442a66a0022c4426a00444a66a0062a666ae68c0e40044cc0a80c00085888585888c8cc0cc88cd400520002235002225333573466e3c0080244c01c0044c01800cccc06dd7280d1aa806111111111100398191119a800a4000446a00444a666ae68cdc780100388008980300199b8148001200202902a026026102d1635573a6ea80384c068d4c064d40040ac0614cd4c06c014840045894cd4c014004854cd40040808854cd40040448809007c4c064d40040954ccd5cd18149aab9d0011323301c357420026ae84d5d10009aab9e0010273754006604844a66a0020384426a00444a66a6604400404a260420022600c006604644a66a0020364426a00444a66a66042004048260400022600c006264a66a6a6aa666ae68c090d55ce80089909180080118079aba135573c00204440024466a004034403c420022c6ea8004888008cc048894cd4008010400405c88d400888d400c88c8cd40148cd401094ccd5cd19b8f00200100301b201f2335004201f25333573466e3c00800400c06c54cd400c854cd400884cd40088cd40088cd40088cd40088cc08000800480888cd400880888cc080008004888088888cd401080888894ccd5cd19b8700600315333573466e1c0140084ccd5cd19b87004001023024020020019153350012019007101423301022333500301b0020013500101a13300175ceb488c88c008dd5800980d111999aab9f00120162323301233013300635573a002600a6aae78004c010d5d10019aba1002019233500100c011223018225335001100322133006002300400123004350010022220032325333573460306aae740044c8c8c8c8c848ccc00401400c008d5d09aba2002357420026ae88008d5d08009aab9e0010163754002464a666ae68c05cd55ce80089919191980618031aba1003533357346034002264646020a666ae68c0740044c8c8c8c8c8488ccc00401801000cdd69aba1357440046eb4d5d08009aba2002375a6ae84004d55cf0010a999ab9a301c001130103008357426aae78008068d55ce8009baa357426aae7800854ccd5cd180d80080680c1aab9d00137546ae84d5d10009aab9e001015375400246464a666ae68c0600044c8488c00400cdd71aba135573c0042a666ae68c05c0044c02cdd71aba135573c00402a6aae74004dd5000919191999aa999ab9a3370e90030008990911118020029aba135573c0042a666ae68cdc3a400800226424444600400a6ae84d55cf0010a999ab9a3017001132122223001005357426aae7800854ccd5cd180b0008990911118018029aba135573c004028401c401c401c4666aa60200226601244a66a00442006200201c46464a66aa666ae68cdc79a80100a9a80080a8999ab9a3370e6a00402c6a00202c02002201a2601c00601a600e006600c6a0020266600e464a666ae68c064d55ce800899198061aba1001357426ae88004d55cf00080b9baa00135300635004012222222222200a35573a0026ea8d40040408c94ccd5cd180a1aab9d001132323300853335734602c6aae740044dd71aba135573c0020286eb4d5d09aba200237546ae84004d55cf0008091baa0012325333573460266aae740044c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c848cccccccccc00404c04403c03402c02401c01400c008d5d09aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026aae78004044dd5000919118011bac001300e2233335573e002401446600a60086ae84008c00cd5d1001006910911980080200191091980080180108021091180100191199ab9a3371e004002006008200a200a9111cd195ca7db29f0f13a00cac7fca70426ff60bad4e1e87d3757fae848400300422112225335001135003008221333500500930040023335530070080050040011200122002220012323001001223300330020020011533573892010350543100162222222222009370e90001b8748009049fd8799fd8799fd8799f581c030c012b604482130d1f821d2ec39f76218cd89c00a3fa7662fbe7b5ffd8799fd8799fd8799f581cef7879c68377187aad9335d004f3abb73dd109f6f674877b367c246affffffffffff0581840001d87980821a002dc6c01a77359400f5a11902a2a269657874726144617461816b4144413a33363536313736636d736781774d696e737761703a204861727665737420726577617264

**- TX Hash: e4f3b69ac3f03b4d2b6aafbf388c11360ce4d0f37a6a7dd141ef78abe162250f**
   Payload: 84a90081825820a18686df5f2b7d3084ce025a863b49c173db2d84eca95437709bed3af6342636010d80019f82583901244517e27015ad768873e1e9709c23d8d53fda9d13ec17df1f3238f246392105de06ad3d071fdbe9cc50626f4791ca8cc416ca1ea51bbf00821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a5fe1c5e233fea6007d4886b01be3551000f376431f8dbabebafa2ea8d6a5e6ff7a50ca95a4334a12eccc6700f16f18e380f2d23c58be195821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019cdb4c2607ffdd6ba63616e88ec99cd6273e6b7bf20c9af9e950a9ee4c6fb1f8b740d2c91eb3e5a4af2fd6802acd08f3e0f2957a7c8ecb7f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390187a25fdc04ecaab290d345a787d82e4593cd7cd4f99f3363cd1da3510216006e6b26a6fbddf7581f5d41a87b024f99a6bce853dfcdec03c5821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839011125eabcaa9f685822c274352333c3c08c4435a1b339253f51c140e5db8ae552b600f7292eaab77dd8e3d68861832637e800d38f23723ce1821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901d359d554b2b3554f630aafb95b9e5e9e414a199c40d753e44e65f01b36441849bf9bf250887d10573bbb735e9b99d828b9d386fe6ba18a14821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016069c7ec26dfb7d3746e2e8a63bfdd92e19f28babc4f74a68ce8065c0cbd41e6e3bc29f4aef19effd143d9e0136a5ea746ab0248d599ee5f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901519448177d96e08f753eb1d6178675def781efd46e8f24f94a4f87ab6b7359005e84705be1c5d37a1cc8d3ab66bb154e422a0c7494284cba821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839017a063499779876c7b0cb972423426612327cf471931bc5b408f6e8ff29b2251206cd6313128d2a36d2657780a10a3b0a139b89954bdee77a821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901030c012b604482130d1f821d2ec39f76218cd89c00a3fa7662fbe7b5ef7879c68377187aad9335d004f3abb73dd109f6f674877b367c246a821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839011febf83bb311d5eb94468a6b623c859007900f6af6da742f6c3e6bd541d7d3aea9abd22161ab887c02243dcaa7b162b4aeed00674f883c98821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016990c1287a2d384768b4a1c42a11bc7834aa3ee3652ea6dde9df1c7df01211dcf6c131892ca2dacdd8eec0f76555adb0483f794c48120279821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019ed24e2cd97760754df81890af75dd835035aec42300b23a3b934e87abc2b3e3dec569c3ef70412cfec6cf8e495e8fbf5215a2805f04a1fc821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839013d9d72c03f18d608aef50703ca176dc7176d88d572d326c6c4a9abb9f6dcac4ce8f5fb286fae54b84ee753813c9ae3738316907f6ad01349821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901531c901b08f2bca0c5bd48da415cac4b20d902d0376b0362483134e4314cd972ebb873d85e1b531620ce4e10f95eeecea891f682778738da821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839017143a88fb0114f01d9cb8ca04f8e1efc68d589b3576d8c00635911feab92ce979955bdaae3ec6fc471c6a196b6b5627770416631475e938d821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901321891ce70ac8366bfdc62ca1faed8c91067c5f530e4bb0ca20ee8b940daffdcae4c99b75baaebe11804794199c1aaa5f78d7a9087818c22821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901b743ce7aef466aecfdc2b6ee5779b83c5b97a64807880ee8e668e6859b12722604cc29bb09d0ef83b783c347583f344ca98c3c4336258c31821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839011ebb98a6ccdaa0449293596ea656f1a9edc3d94b8bac9a4d2af2d6bc1050a0317ff2aaf1eb38d3029a8716acf697baa899f0977f98945da0821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a9c8a5205ecd62fced4dfe740970aaa75a086c18252891e6dc95e5a1c980c8cc0bbcdfcde4b02abab576538f1c5965dce41d95eaa509aef2821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390129bf850a2fd920a5216cc8ed933ca19859eded035f42f9491091a421d14fa656799fb7332626067a70033205ee42759bc652b43af82ed969821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901894a61ac5c2d56eb5994c6c827c3a1ae4438006a909e80fdf5969da44481688c216a80e0d55080113123eb542ad76661302bf52bb76fbe61821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016926442f03a87d9801ae52e693c5e77292d70c86376e55c0c6d647ddd4e4eb045a8d4daf2f85ab337e495721c5e8cb5d520932b75d5886e3821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839018b573314014c1516d15fa87ee8b016b4e7f554a14b059457b4279e02b369fc0a215f19e872a718911357292485d7a1cca2c4ca9c2ba96a90821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390198c7351a5425c8ef0c6ba9a9f7338863efa387c5dbeca109d012044750243b5ab0fac66e4d96d3ab666b60ff6869cbb9eca1a0e8949fc016821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901ee598da405bf697d45f77498b480a3653ae599fd00e3844cd95fb8c2498d81efc3eb4f279e7936351d7c8d5bf9c279afb1188c6917b3508f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390123a7d8c1fd5c69a828eee2be1206e4c5616203bab205b0e6c434ec6fbd3765599fa308144b74f9f900f9db7f9f6871098bac3fbdb7e631a8821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901e1d687de49715a4b39e4a9deccf53c37e9bfd5bd71864bc2a07efe1ef061d46ab15f879f494d1cda49d151d0b76c159440dfd5a9a8f6b0db821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839015f55083f88eeccd16c23abbfa320cb910bea1461fd0b7bc9bd754baa30a4ec1b59c8a7915c50f302b08d6e82e54e653bd8964e3052a9db02821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901e9ca7b0ecdb5a6ca7aeaed865357d442238f0769e45120d40b56544ff3cf3c0e08c7a5a1f96db8167518bcf05201356855f444b0a0e8c3d6821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019b8c8e1e5f4a7db4be9865a6fe5eb2f33cf843383b0177d333f063eb0ad9ca8a5d6a6193bc9b0d3f3fe488f6389f45f92cf7760cf4cc7b29821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390101db473914c51ed94cf2635b2560ef81d8e1ee9a2f04b6739bf0c6b514f80981990a97792a9aff38d095dcaf69aa8f954f181b4ecccd8038821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390134b6850a8f0d4b201db962896ad1e8354b0e3c00394b376c09dc6bf045d6659823a4f4d6fb9397ed6c8d244fe4e72f94f082f9a4d2aefb97821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019e0575a93f6a4f10c3c1fc861351fa505f4b82db92da90557d77322e3c44457c0a64f602af6cd4d643c6087874a10db92422f71b03a26dc4821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839012f6b3a3743263e2acaa4c597fdef505baebc0cbbcacc44de9c1477719ee0e4ca785e27ee54df38d61a330e96b21ad6bbeb2d9388d8f478fc821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390134e6b3b08b033ef4cecb32ecb37f71d441ab3a22191db7d42eae6fcffc3d0ad63395ca631a23e0ead2634ee1389255312f908ec366737e72821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901e29fe12b4456ce5fe5c7c9b3f83708c36b9e9bdf65fd311233a186134cc047f79d1dd30dd72ab7e731169f9bf5af7524c7328f1b8efcfd26821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016654216855f808db76a7ae22bf2b4fcd5f454526a391bec10fa73f292789f369635ce3f8ced4a96d902446a447951a4fa9d63d009055cdc8821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a85a30bf0f0b3bbf4f3a8841b87c417dda45197e05331696aa92118cba52cc06715b7c6522298aeaabf4f26d76c6a489085c03c756816749821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901dc7430783b8008df7bae1aac9a54e600be105b8b092a2c5554695b041068dbb11fd3a169950fd1f1c8122abc61df35d5a60cae6b3c7f593f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839013cd56ddf10c9b194ec02f60ca45a07e8ce92e1b21ee97a9944d3a7a2e927fcce28d861cf7fec2589436750b91ac3db47e510a8eeb6d3e045821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901011fcfe32abce4f2359a5008f8680ee41e115c093a09cde1bcb7601653e866ffe127c2683e17fa217acee3fc4a4f452d97f88d37210ea174821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019501070154372d7e520773bfd9491878faea8db5e97e51dab05a302c86a1cef34e9f361e6a587f248c7684d5d939c6e17b8eef76932f2750821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901890e665602a3e0abcfa998fdc033f6887302c62001a442eb1ea25e91951988d3a0d000c13566d252de2eb24a06a24f6f744a9a26f587d77f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390127f08e2992f5072ecd4f4bb7f0a5fc9ba644e41cdb37d894e912946107c81f180cf6018d0689c062b2775a801c356c019ed10b77cdbf2a81821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016674949af14f28ab2a707421a4d6052cd52171a7b2cf231a5c7fde678bf43a42917a308a0a6c455abaf5ed383cceb08d655615f67f29763e821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901f381c09bf167609b8149ce5f3f9268283bec58f3fae60c0b3eff5d110b1323cd0a53e130f0349452c4e8d49c77d0f4d5f3b445ceab3c3bf2821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901101405f98961e9effeccfb4c9a7690fb7873d3023165cff06179fda69f17d46175c56b7853b71a3bd611938b4588dd594847f1f135f15cad821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901ef277349610171bba6a6aa284a907311f21c164d84665310721b00bc9809bd072301d43403fdbb272f65ee3973c7455bddf8b5a55ed93586821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839010ab8ff1f0a19c40293fc87ceb9d8fbae8e81bc66efce4c8c93c4a1cd2c58a45008a136f8356942bdd2d1d8f582a06681e0a1ebfb9bc5058c821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901eba8f48197291ce3d3188bb0c2152512dc7166cbfe8b7d822c52072e6e31a5c3a286226db62d72b367a7d02179450f505bc2afa1dd9c58d5821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a95bd2e5b8add96efc37e3fd67cc7460b0e4613e2646e3e9ad5bf76bb4f4e9544753bd53a7e5d60345aa4d5e7143967a2efb88298fe2d29b821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a5dcdbf8067f1c11754435dee189ce90d50b11822c8bb2de89f2281edbb1385cbaf294cbbeda70b2f0b37d5a4d2433b6115ac93ce0eb6f46821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839019119b09bd0ed6126de796c5b758ea8447694164f2666914557806a52a0d8fd05f0936e1e75799983a592fb4c7362e03f645b03ab3611e351821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901370e5b426f929b443de3cc411e166bc60df050e5a1af4c7087e06eb34abf4d97157feb01fa89edc859a63889957559d1d1e9e823cb728bbf821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016d19ca9bf9e71ce9f7b19b2ddd1a8908e23b8a12ad57641223ba4a6e6c60672d5c1a4cab22b3235ec4dbbbae24c963ae4b14b945bc8e6468821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901c7b97f36adc7691fadc2edeaf6270cad6cd53c82161e00c8f933f2661b0defad58b2d8a7c6009e54c034f44a05aceee0cc0bff7990193295821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a63a6f4402d87b1abeb25703de4e55a0b3cdd2aa07b92ced59b3ec54594f981c3c1534c3fedaf8a36630b2af8dc45cc407e5d780c998e8b5821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901ee9da565a7bb342b9626875a33687263ca9b9421e91384b23bb2dd67ce70b47c224110c1c3624bf35345094583bca6b9def95b520c54d95f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a3fe41e87804c8a6c5e8f6ee9ebfae7cdec7ed41d7fd86d07b19ae911630f135e0e3b4508387adc0c9a706f60167f4016514784efbe462db821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901fb4d5adf4ccd126d68aaa5b3ca9f6e91c5e604f5117540c7a102dce2ec1f717f97dda991c7e6d97b71bdf83f33daeb6bf93c121350c3e484821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901c84f689e5b218082389a556b725e2396ff2dfe378eb00e64608e7fdc1d73afa34d7ec8c412a256e956eb434793268c63d5500044a042870c821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901858ded85b9518aa42de4eb88d897cabba8980484e3d9e1266258259464971decdd8523cb3b8bdedaf5f1bf7a672f3ba38f59b87297531494821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901a358e486b4a829dab93113616d1188cbedf0b60231d47d663c70d02b0330c6c82f4d5b1c9d9ae6142792ba0e1a91b0cd8e193b00bf1f682c821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901b4617decd0f96ee313c76e9849ca171b4d3dfb2c375cca8af1c0bfaf1b5af1f3c29ec2f03ae725719d829dfed8fbbd60725c8581538b8e2f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839018e01366a34374820bc772a6c1e01a12af3c10865a944dc1c3dbdc05c7c53f0c1382d76d597339c46a6e5f4658c60d50e6a17da13b6af23fa821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182583901fa0e6de8b45ea6b1ee2c818348b9cd1e9c6edb782d7f3ab10401fc67cdb29b219d44be2b9c31a66726b1a75da54a6a8d1072fd99f7df045f821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e46545768697465018258390136234388dd48387ea667860aa613c7efc7ccef7dbdb8123e10e592721029a2ae73d85306d7c74f817b6dfc18946b722d9ea1033450b3b83b821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839016543034cd5553ae6f0feac02047690c47770cc21033b42abe6b0099d2ff566332b4a1fd7e0074b6b487a45b6105c93be76ad65bc813203cb821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e4654576869746501825839012dbdd0d407a7b53cb363cc1a0f3565e82f3fb628bdf7f9d69b7b9125af9172e7784771d1ac5479167db0fbef9c6713230fb19069484fd7c4821a0015cf20a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974650182581d61e4d3489bc0bdbdebcf5852ffd743b1c5bdf09c6b2aa0c6fcca2b28561a07089ba7ff021a000e6e59031a04d36d2d081a03e98f640e8009a1581ccb6989d37ab131053737ccab9d22720880a07276eb04e137a4e58291a14c565946494e465457686974651846075820684f55974d82d832ccce95e1bf39f994eba3add4f8de241809f8ea4035f33b84a200828258201f2a35ef7116a5fa7b5d0faa074740b3b44c7ea0905c795f95718ea2c01c56c25840ad7db3dc707b769fed2da4d93bd3e9f928c206433721ad6527bada96560d3df2e4bb4a091ad1d6ee11e95f4955f5b6e22b46ae1905fe2c9e1fa5971123e34106825820184936daafef96bf0c2e799a9e2b9f095ffb70a3c4ccc8f9381e0f760702e9ce5840bea7abcd1aeba09b16e421d16a7728dfeb8f22cbcd508ab8c9fa022ccf447ab612a94507af7d3acf99f9d3daa20800981e917838cee2151dc9c1feb199cad408018182018382041a03e98f6482051a04d36d2d8200581c0d9b39474eb05f95af7ea93ac6bcf3e4fcb64159a1eb0823efe85824f5d90103a100a11902d1a278386362363938396433376162313331303533373337636361623964323237323038383061303732373665623034653133376134653538323931a16c565946494e46545768697465a56b6465736372697074696f6e78235374616b652026204561726e202d2068747470733a2f2f6170702e767966692e696f2f6566696c657381a3696d656469615479706569696d6167652f676966646e616d656c565946494e46545768697465637372637835697066733a2f2f516d574756745634526f6e34745133527374315947366a637064484d5876537853427445716343566d365369636d65696d6167657835697066733a2f2f516d574756745634526f6e34745133527374315947366a637064484d5876537853427445716343566d365369636d696d656469615479706569696d6167652f676966646e616d656c565946494e465457686974656776657273696f6e63312e30

**Comment by satran004** [comment]
@stake1edge 
It seems Minswap code is using indefinite-length array during serialization where the last item is BREAK.

To handle this scenario, cardano-client code has to ignore BREAK during parsing.

I did a quick test. The first one has  indefinite-length array for plutus data and the second sample has indefinite-length for tx output array.

**Comment by satran004** [comment]
The current parsing code is https://github.com/bloxbean/cardano-client-lib/blob/06a48ea352a9cc3c52253cc9a190c198a0180058/src/main/java/com/bloxbean/cardano/client/transaction/spec/TransactionWitnessSet.java#L210

```
for(DataItem plutusDataDI: plutusDataDIList) {
        plutusDataList.add(PlutusData.deserialize(plutusDataDI));
}
```

After fix, it should look like

```
for(DataItem plutusDataDI: plutusDataDIList) {
        if (plutusDataDI == SimpleValue.BREAK)
                    continue;
        plutusDataList.add(PlutusData.deserialize(plutusDataDI));
}
```

But the same error can occur for other arrays like redeemers, txoutputs if an indefinite-array is used. I think it's better to handle this scenario for all arrays in transaction. But we need some more sample transactions to verify that.

**Comment by stake1edge** [comment]
Ok, heres a few more example tx:
e4aa8e05ed1dadc6e606672e85da8bb38a74119b001449e49f2b8cf1d6bce8c1
37370bbe79e62dc2b2fdc738a2a721e99a1c341e6f3ae85c0a256b70f039320d

TxOutput : SPECIAL error related:
b9602c155872d6adef89eb6764c432b27614147f679994fadcaaaa2fa9c9acfe
235b1710e66a713b8be45e4c5c1fb835a3afbf370cf62717759a85473ae20215

Do you just need the tx id?

Can provide more examples that I think are similar to these categories. I think I did also find another issue deserialising some random utxos, will see if I can regenerate.

**Comment by satran004** [comment]
Thanks @stake1edge

It would be great if you can add txbody  in a file and attach here.

Otherwise, let me know if there is an easy way to get the original body from tx id without going through the chain data. Any explorer exposes the original cbor ?

**Comment by stake1edge** [comment]
ah ok no probs attached a file, it includes the original two tx.

[java-client-lib-example-tx.txt](https://github.com/bloxbean/cardano-client-lib/files/9553783/java-client-lib-example-tx.txt)

The other problem I found was something different, will post separate issue

**Comment by satran004** [comment]
Fixed in 0.3.0. @stake1edge, feel free to re-open this if you still see this issue.
